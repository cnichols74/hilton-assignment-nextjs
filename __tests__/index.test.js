import { shallow } from 'enzyme'

import App from '../pages/index.js'

describe('Tests for main entry point.', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<App/>)

    expect(component).toMatchSnapshot()
  })
  it('should render banner text correctly with given strings', () => {
    const strings = ['one', 'two']
    const component = shallow(<App list={strings}/>)
    expect(component).toMatchSnapshot()
  })
})
