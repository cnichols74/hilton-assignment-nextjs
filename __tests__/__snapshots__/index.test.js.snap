// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Index correctly. 1`] = `
<Fragment>
  <main
    className="jsx-2994941223"
    role="main"
  >
    <div
      className="jsx-2994941223 flex-container"
    >
      <Rooms
        change={[Function]}
        click={[Function]}
        rooms={
          Array [
            Object {
              "adults": Array [
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "children": Array [
                Object {
                  "displayValue": "0",
                  "value": "0",
                },
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "id": 1,
              "isSelected": false,
              "selectedAdults": 0,
              "selectedChildren": 0,
            },
            Object {
              "adults": Array [
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "children": Array [
                Object {
                  "displayValue": "0",
                  "value": "0",
                },
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "id": 2,
              "isSelected": false,
              "selectedAdults": 0,
              "selectedChildren": 0,
            },
            Object {
              "adults": Array [
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "children": Array [
                Object {
                  "displayValue": "0",
                  "value": "0",
                },
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "id": 3,
              "isSelected": false,
              "selectedAdults": 0,
              "selectedChildren": 0,
            },
            Object {
              "adults": Array [
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "children": Array [
                Object {
                  "displayValue": "0",
                  "value": "0",
                },
                Object {
                  "displayValue": "1",
                  "value": "1",
                },
                Object {
                  "displayValue": "2",
                  "value": "2",
                },
              ],
              "id": 4,
              "isSelected": false,
              "selectedAdults": 0,
              "selectedChildren": 0,
            },
          ]
        }
      />
    </div>
  </main>
  <Button
    label="Submit"
    onClick={[Function]}
  />
  <JSXStyle
    id="2994941223"
  >
    .flex-container.jsx-2994941223{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-flow:row wrap;-ms-flex-flow:row wrap;flex-flow:row wrap;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;-webkit-align-content:flex-start;-ms-flex-line-pack:start;align-content:flex-start;}.flex-container.jsx-2994941223:after{margin:5px;}
  </JSXStyle>
</Fragment>
`;
